packages/backend-app-api/src/config/config.ts

```diff
@@ -30,19 +30,33 @@ import type { Config, AppConfig } from '@backstage/config';
 import { getPackages } from '@manypkg/get-packages';
 import { ObservableConfigProxy } from './ObservableConfigProxy';
 import { isValidUrl } from '../lib/urls';
+import path from 'path';
 
 /** @public */
 export async function createConfigSecretEnumerator(options: {
   logger: LoggerService;
   dir?: string;
+  entry?: string;
   schema?: ConfigSchema;
 }): Promise<(config: Config) => Iterable<string>> {
-  const { logger, dir = process.cwd() } = options;
+  const {
+    logger,
+    dir = path.resolve(
+      process.argv[1].replace(
+        // Strip the entry point path.
+        new RegExp(`${options?.entry ?? 'src/index'}$`),
```

Comment:
Bit to risky to be doing string manipulation here imo, thinking it's best to stick to path operations instead if possible? Could be that we want to do something similar to what `@backstage/cli-common` does with scanning for the closest parent `package.json` file or something like that?

Reply:
Moved to getting the closest package.json in [b39977a](https://github.com/backstage/backstage/pull/23661/commits/b39977a57de404615465a9fd75028a6469302d10)